packages/strapi-database/lib/queries/helpers.js

```diff
@@ -9,9 +9,12 @@ const withLifecycles = ({ query, model, fn }) => async (params, ...rest) => {
   const queryArguments = [newParams, ...rest];
 
   // execute before hook
-  await executeBeforeLifecycle(query, model, ...queryArguments);
+  const populate = await executeBeforeLifecycle(query, model, ...queryArguments);
```

Comment:
This could be an array so we can map the return value with the argments and all overwriting any of them instead

Reply:
Sorry, I don't get you. What could be an array?

Reply:
There is not reason to return only a populate argument. all the beforeLifecycles don't have one. We could allow changing input arguments by returning an array like so

```js
const returnedArguments = await executeBeforeLifecycle(....);

if (isArray(returnedArguments)) {
  returnedArguments.forEach((arg, idx) =>  {
    if (!isNil(arg)) {
      queryArguments[idx] = arg;
    }
  })
}
```